Frontend_Web_Development_React_Angular_Vue/Vue/Week 5/Emitting_Events.md

```diff
@@ -0,0 +1,201 @@
+
+Unlike components or props, event names don't provide any automatic case transformation. So the event names should be the same where the event is listened to.
+
+The most common ways to use emit is:
+
+1. Inline using `$emit`
+
+2. Options API - `this.$emit`.
+
+We will see the working of the above two syntaxes in a while.
+
+## Advantages of emitting events:
+
+1. Passing data from any component to another with the help of the `$emit` payload object.
+
+2. Overriding the native events to achieve many concise and complex actions to be performed as and when required.
+
+3. Building wrappers for event-heavy elements like the `input`, `form` components.
+
+## Need for emitting events:
+
+It is generally used to notify the parent component that something has changed, and also can be used to send data while an event is being emitted to change/update its state or value, or compute some fields which might require that data value.
+
+## Examples
+
+Child Component: `Hello World.vue`
+
+```html
+<template>
+  <div class="hello">
+    <input type="text" placeholder="Custom events demo" v-model="color" />
+    <button @click="$emit('custom-event', color)">Submit</button>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: "HelloWorld",
+    props: {
+      msg: String,
+    },
+    data: () => ({
+      color: "Enter the background color",
+    }),
+  };
+</script>
+```
+
+In the above component, we are emitting an event `custom-event` when the button is clicked after entering the value in the input box, then the above value is passed to the parent component as shown value, which hears for it and updates the background color to the same as the value in the input box.
+
+Parent Component: `App.vue`
+
+```html
+<template>
+  <div id="app">
+    <HelloWorld @custom-event="changeColor" />
+  </div>
+</template>
+
+<script>
+  import HelloWorld from "./components/HelloWorld";
+
+  export default {
+    name: "App",
+    components: {
+      HelloWorld,
+    },
+    methods: {
+      changeColor(event) {
+        document.getElementById("app").style.background = event;
+      },
+    },
+  };
+</script>
+```
+
+The changeColor method changes the background of the element with id `app` to the value passed in the emit event. Here we are using `inline-emit`.
+
+The above code in action:
+
+<div style="display: flex; justify-content: center; align-items: center">
+  <img src="./assets/inline-emit.gif" style="width: 500px;">
+</div>
+<br>
+
+The same can be achieved with the Options API using the `this` keyword as shown below:
+
+```html
+<template>
+  <div class="hello">
+    <input type="text" placeholder="Custom events demo" v-model="color" />
+    <button @click="$emit('custom-event', color)">Submit</button>
```

Comment:
here as per your heading, you should call "emittingEvent" method with on click of the button.